src/DIRAC/FrameworkSystem/scripts/dirac_login.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python
+########################################################################
+# File :    dirac-login.py
+# Author :  Andrii Lytovchenko
+########################################################################
+"""
+Login to DIRAC.
+
+Example:
+  $ dirac-login -g dirac_user
+"""
+from __future__ import division
+from __future__ import absolute_import
+from __future__ import print_function
+
+import os
+import sys
+
+import DIRAC
+from DIRAC import gLogger, S_OK, S_ERROR
+from DIRAC.Core.Base import Script
+from DIRAC.Core.Utilities.DIRACScript import DIRACScript
+from DIRAC.Core.Security.ProxyFile import writeToProxyFile
+from DIRAC.Core.Security.ProxyInfo import getProxyInfo, formatProxyInfoAsString
+from DIRAC.Resources.IdProvider.IdProviderFactory import IdProviderFactory
+from DIRAC.FrameworkSystem.private.authorization.utils.Tokens import writeTokenDictToTokenFile, readTokenFromFile
+
+__RCSID__ = "$Id$"
+
+
+class Params(object):
+
+  def __init__(self):
+    self.proxy = False
+    self.group = None
+    self.lifetime = None
+    self.provider = 'DIRACCLI'
+    self.issuer = None
+    self.proxyLoc = '/tmp/x509up_u%s' % os.getuid()
```

Comment:
What is this for?

Reply:
to get a proxy certificate instead of a token, I will add the keys

Reply:
Sorry, I am confused then: isn't `dirac-proxy-init` for the proxy, `dirac-login` for the token?

Reply:
`dirac-proxy-init` use user certificate to generate proxy with DIRAC group extension.
`dirac-login` use DIRAC authorization server `authorization_endpoint` to AuthN through IdP(IAM/CheckIn) and AuthZ:
  - by obtaining access tokens with scopes that related to DIRAC group from IdP(IAM/CheckIn).
  - by obtaining proxy with DIRAC group extension from DIRAC `ProxyManager`

Reply:
I'll think about how to combine it all.